docs/localdev.md

```diff
@@ -86,3 +86,39 @@ We will address two modes of working with the ExternalDNS operator
 ### Working with external Kuberenets/OpenShift clusters
 
 If you have any other cluster you'd like to test the operator on you just need to set KUBECONFIG and follow the instructions layed out in both sections indicated above
+
+## Kind (Kubernetes in Docker)
+This section walks you through the deployment of the external-dns-operator in a Kind environment.
+### Prerequisites
+- [Go](https://golang.org/doc/install#install) and [Docker](https://docs.docker.com/engine/install/) are installed.
+- [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) is installed and a local Kubernetes [cluster](https://kind.sigs.k8s.io/docs/user/quick-start/#creating-a-cluster) is created.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) is installed to access the above created cluster.
+- Ensure the above created cluster is accessible by any `kubectl` command.
+- If the cluster is not accessible by the `kubectl`, excute the following steps to point kubectl to the configuration that the kind is using.
+   ```bash
+  sudo docker exec $(sudo docker ps -a | grep kindest | awk '{print $1}') "/bin/sh" "-c" "cat /etc/kubernetes/controller-manager.conf" > ~/.kube/kind-config
+  export KUBECONFIG="~/.kube/kind-config"
+   ```
+- Any registry that is accessible from your cluster to store and distribute the docker images. [quay.io](https://quay.io/) is shown as an example here.
```

Comment:
Why do we need to pass by a registry?
`kind` has a command to load images from the host:
```
$ kind load docker-image -h
Loads docker images from host into all or specified nodes by name

Usage:
  kind load docker-image <IMAGE> [IMAGE...] [flags]
```

Reply:
Not sure I understood the link between `imagePullPolicy` and usage of the external registry. `kind load docker-image` places the image on the nodes like if they were pulled by `docker pull`. You need to do a rollout of the deployment afterwards, same as if a new version of the image was pushed to the external registry.

Reply:
As we found out, the problem is on the `imagePullPolicy` (as previously stated by @sairameshv ). From Kind's [doc](https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster):

> NOTE: The Kubernetes default pull policy is IfNotPresent unless the image tag is :latest or omitted (and implicitly :latest) in which case the default policy is Always. IfNotPresent causes the Kubelet to skip pulling an image if it already exists. If you want those images loaded into node to work as expected, please:
> don't use a :latest tag
> and / or:
> specify imagePullPolicy: IfNotPresent or imagePullPolicy: Never on your container(s).

I'm fine with using of `IfNotPresent` image pull policy for the operator deployment. It would make it more production ready.